test/conformance/api/v1/volumes_test.go

```diff
@@ -402,3 +404,71 @@ func TestProjectedComplex(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+// TestProjectedServiceAccountToken tests that validates the default service token mounted
+func TestProjectedServiceAccountToken(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "hellovolume",
+	}
+
+	const tokenPath = "token"
+	saPath := filepath.Join(filepath.Dir(test.HelloVolumePath), tokenPath)
+
+	withVolume := WithVolume("projectedv", filepath.Dir(test.HelloVolumePath), corev1.VolumeSource{
+		Projected: &corev1.ProjectedVolumeSource{
+			Sources: []corev1.VolumeProjection{{
+				ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
+					Audience: "api",
+					Path:     tokenPath,
+				},
+			}},
+		},
+	})
+	withSubpath := func(svc *v1.Service) {
+		vm := &svc.Spec.Template.Spec.Containers[0].VolumeMounts[0]
+		vm.MountPath = saPath
+		vm.SubPath = filepath.Base(saPath)
+	}
+
+	withRunAsUser := func(svc *v1.Service) {
+		svc.Spec.Template.Spec.Containers[0].SecurityContext = &corev1.SecurityContext{
+			RunAsUser: ptr.Int64(0),
```

Comment:
Let's add a comment here to say why it needs to run as root? Something like:

```suggestion
                        // The token will be mounted owned by root, so we need the container to
                        // run as root to be able to read it.
			RunAsUser: ptr.Int64(0),
```

Reply:
fixed.